engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -87,7 +86,8 @@ public final class EnsoContext {
   private final boolean isPrivateCheckDisabled;
   private final boolean isStaticAnalysisEnabled;
   private final boolean isHostClassLoading;
-  private final boolean isResolverClassLoading;
+  private final boolean isGuestClassLoading;
+  private volatile Object guestJava = this;
```

Comment:
Why `volatile`?

Reply:
Refactored, moved away and made `synchronized` at the end.